src/gesture-handler/GestureDetector.tsx

```diff
@@ -0,0 +1,302 @@
+      global._manageScreenTransition(
+        ScreenTransitionCommand.Finish,
+        stackTag,
+        true
+      );
+      return;
+    }
+    transitionConfig.screenDimensions = screenSize;
+    startScreenTransition(transitionConfig);
+    canPerformUpdates.value = true;
+  }
+
+  function onUpdate(event: GestureUpdateEvent<PanGestureHandlerEventPayload>) {
+    'worklet';
+    if (!canPerformUpdates.value) {
+      return;
+    }
+    if (goBackGesture === 'swipeRight' && event.translationX < 0) {
+      event.translationX = 0;
+    } else if (goBackGesture === 'swipeLeft' && event.translationX > 0) {
+      event.translationX = 0;
+    } else if (goBackGesture === 'swipeDown' && event.translationY < 0) {
+      event.translationY = 0;
+    } else if (goBackGesture === 'swipeUp' && event.translationY > 0) {
+      event.translationY = 0;
+    }
+    let progress = 0;
+    if (goBackGesture === 'swipeRight') {
+      const screenWidth = screenTransitionConfig.value.screenDimensions.width;
+      progress =
+        event.translationX /
+        (screenWidth - startingGesturePosition.value.absoluteX);
+    } else if (goBackGesture === 'swipeLeft') {
+      progress =
+        (-1 * event.translationX) / startingGesturePosition.value.absoluteX;
+    } else if (goBackGesture === 'swipeDown') {
+      const screenHeight = screenTransitionConfig.value.screenDimensions.height;
+      progress =
+        (-1 * event.translationY) /
+        (screenHeight - startingGesturePosition.value.absoluteY);
+    } else if (goBackGesture === 'swipeUp') {
+      progress = event.translationY / startingGesturePosition.value.absoluteY;
+    } else if (goBackGesture === 'horizontalSwipe') {
+      const screenWidth = screenTransitionConfig.value.screenDimensions.width;
+      progress = Math.abs(event.translationX / screenWidth / 2);
+    } else if (goBackGesture === 'verticalSwipe') {
+      const screenHeight = screenTransitionConfig.value.screenDimensions.height;
+      progress = Math.abs(event.translationY / screenHeight / 2);
+    } else if (goBackGesture === 'twoDimensionalSwipe') {
+      const screenWidth = screenTransitionConfig.value.screenDimensions.width;
+      const progressX = Math.abs(event.translationX / screenWidth / 2);
+      const screenHeight = screenTransitionConfig.value.screenDimensions.height;
+      const progressY = Math.abs(event.translationY / screenHeight / 2);
+      progress = Math.max(progressX, progressY);
+    }
+    sharedEvent.value = event;
+    const stackTag = screenTransitionConfig.value.stackTag;
+    global._manageScreenTransition(
+      ScreenTransitionCommand.Update,
+      stackTag,
+      progress
+    );
+  }
+
+  function onEnd(event: GestureUpdateEvent<PanGestureHandlerEventPayload>) {
+    'worklet';
+    if (!canPerformUpdates.value) {
+      return;
+    }
+    const screenSize = screenTransitionConfig.value.screenDimensions;
+    const distanceX = event.translationX + event.velocityX * 0.3;
```

Comment:
Link to where those values were taken from.

Reply:
it is empirically determined :/